File: Binary_Search/BOJ_1920_Find_Number.cpp
#define _CRT_SECURE_NO_WARNINGS

#include <cstdio>
#include <algorithm>
#include <vector>

using namespace std;

vector<int> og_list;

bool bin_search(int num, int start, int end)
{
	int mid = (start + end) / 2;
    
    if(start > end)
        return false;
    else
    {
	    if (num == og_list[mid])
		    return true;
	    else if (num < og_list[mid])
	    	return bin_search(num, start, mid - 1);
	    else if (num > og_list[mid])
		    return bin_search(num, mid + 1, end);
    }
}

/* BOJ 1920 */
int main() {
	int N, M;
	int temp;

	/* USER INPUT */
	scanf("%d", &N);

	for (int i = 0; i < N; i++)
	{
		scanf("%d", &temp);
		og_list.push_back(temp);
	}

	sort(og_list.begin(), og_list.end());

	scanf("%d", &M);

	for (int i = 0; i < M; i++)
	{
		scanf("%d", &temp);
        if(bin_search(temp, 0, N - 1))
		    printf("1\n");
        else
            printf("0\n");
	}

	/* Processing */


	/* PRINT RESULT */

	return 0;
}

File: Binary_Search/BOJ_1920_Find_Number.py
n = int(input())
arr = list(map(int, input().split()))
m = int(input())
find = list(map(int, input().split()))

arr.sort()

for target in find:
    
    finded = 0
    start = 0
    end = len(arr) - 1

    while start <= end:
        mid = (start + end) // 2

        if arr[mid] < target:
            start = mid + 1
        elif arr[mid] > target:
            end = mid - 1
        else:
            finded = 1
            break

    if finded == 0:
        print("0")
    else:
        print("1")